File: R/l1median_VaZh.R
l1median_VaZh <- 
function (X, maxit = 200, tol = 10^-8, zero.tol = 1e-15, trace = 0, m.init = apply (X, 2, median), ...)
{
	if (class (X) != "matrix")
	{
		if (class (X) == "data.frame")
			X = as.matrix(X)
		else
			X = matrix(X, ncol = 1)
	}
	
	if (length (m.init) != ncol (X))
		stop (paste ("length of vector m.init (=", length (m.init), ") does not match the number of columns of data object X (=", ncol (X),")", sep = ""))

	ret = .C ("l1Median_VZ", PACKAGE="pcaPP", 
		npar = as.integer (c(dim (X), maxit, 0, trace, 0)),
		dpar = as.double (c (tol, zero.tol)),
		as.double (X),
		med = as.double (m.init)
		)

	return (list (par = ret$med, value = sum (sqrt (colSums ((t(X) - ret$med)^2))),
		 code = ret$npar[4], iterations = ret$npar [3], time = ret$npar[6]))
}

File: matlab/l1median_VaZh.m
function [medOut, code, niter] = l1median_VaZh (X, maxiter, tol, zerotol, trace, medIn)

    code = 0 ;
    niter = 0 ;

    if (nargin > 6)
        error ('Too many input arguments.') ;
    elseif (nargin < 6)
        medIn = median (X) ;
        if (nargin < 5)
            trace = 0 ;
            if (nargin < 4)
                zerotol = 1e-15 ;
                if (nargin < 3)
                    tol = 1e-8 ;
                    if (nargin < 2)
                        maxiter = 200 ;
                        if (nargin < 1)
                            error ('Data matrix X is missing.') ;
                        end
                    end
                end
            end
        end
    end

    if (size (X, 1) == 1)
        medOut = X ;
    elseif (size (X, 2) == 1)
        medOut = median (X) ;
    else
        par = [maxiter, trace, tol, zerotol] ;
        dParOut = pcaPP (1, X, medIn, par) ;
        medOut = medIn ;

        code = dParOut(1) ;
        niter = dParOut(2) ;
    end
